### PR TITLE
Fix global expression (EXPERIMENTAL & DO NOT MERGE)

### PR DESCRIPTION
**Pull request by lucasccordeiro**

### CODE REVIEW

**Comment by lucasccordeiro** [comment]
```
<benchmark tool="esbmc" timelimit="15 min" memlimit="15 GB" cpuCores="1">
```

```
if concurrency:
  esbmc_dargs += "--no-por --context-bound 2"
  esbmc_dargs += "--no-slice "
```

Master:

```
  correct:             966
    correct true:      181
    correct false:     785
  incorrect:            17
    incorrect true:     12
    incorrect false:     5
  unknown:             147
  Score:               683 (max: 1413)
```

This PR:

```Statistics:           1130 Files
  correct:             967
    correct true:      182
    correct false:     785
  incorrect:            18
    incorrect true:     13
    incorrect false:     5
  unknown:             145
  Score:               653 (max: 1413)
```

ESBMC used to produce timeout for this benchmark `pthread/singleton.c`, but now it provides incorrect true.

**Comment by lucasccordeiro** [comment]
```
<benchmark tool="esbmc" timelimit="15 min" memlimit="15 GB" cpuCores="1">
```

```
if concurrency:
  esbmc_dargs += "--no-por --context-bound 3"
  esbmc_dargs += "--no-slice "
```

Master:

```
Statistics:           1130 Files
  correct:             793
    correct true:      159
    correct false:     634
  incorrect:            20
    incorrect true:     10
    incorrect false:    10
  unknown:             317
  Score:               472 (max: 1413)
```

This PR:
```
Statistics:           1130 Files
  correct:             965
    correct true:      180
    correct false:     785
  incorrect:            16
    incorrect true:     10
    incorrect false:     6
  unknown:             149
  Score:               729 (max: 1413)
```

**Comment by lucasccordeiro** [comment]
This PR: 

```
<benchmark tool="esbmc" timelimit="15 min" memlimit="15 GB" cpuCores="1">
```

```
if concurrency:
  esbmc_dargs += "--no-por --context-bound 4"
  esbmc_dargs += "--no-slice "
```

```
Statistics:           1130 Files
  correct:             962
    correct true:      178
    correct false:     784
  incorrect:            19
    incorrect true:     13
    incorrect false:     6
  unknown:             149
  Score:               628 (max: 1413)
```

File: regression/esbmc-unix5/github_148_fail/code.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+    else if(is_array_type(it))
```
Comment (mikhailramalho) [inline regression/esbmc-unix5/github_148_fail/code.cpp]: Thrash?

File: regression/esbmc-unix5/github_148_fail/main.c.orig [hunk 2]
```diff
@@ -0,0 +1,50 @@
+extern int __VERIFIER_nondet_int(void);
```
Comment (mikhailramalho) [inline regression/esbmc-unix5/github_148_fail/main.c.orig]: Thrash?

File: regression/esbmc-unix5/github_148_fail/test.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (mikhailramalho) [inline regression/esbmc-unix5/github_148_fail/test.sh]: Thrash?

File: regression/esbmc/19_time_var_mutex_true-unreach-call/test.desc [hunk 4]
```diff
@@ -1,4 +1,4 @@
-THOROUGH
+CORE
```
Comment (mikhailramalho) [inline regression/esbmc/19_time_var_mutex_true-unreach-call/test.desc:1]: the test case is name true-unreach-call, are we sure this is supposed to fail?

**Review by mikhailramalho — CHANGES_REQUESTED** [review]
small nitpicks

File: regression/esbmc-unix5/github_148_fail/code.cpp (hunk 1 above)
Comment (lucasccordeiro) [reply regression/esbmc-unix5/github_148_fail/code.cpp]: Fixed.

File: regression/esbmc-unix5/github_148_fail/main.c.orig (hunk 2 above)
Comment (lucasccordeiro) [reply regression/esbmc-unix5/github_148_fail/main.c.orig]: Fixed.

File: regression/esbmc-unix5/github_148_fail/test.sh (hunk 3 above)
Comment (lucasccordeiro) [reply regression/esbmc-unix5/github_148_fail/test.sh]: Fixed.

**Comment by lucasccordeiro** [comment]
This PR is already obsolete.

**Pull request closed without merging**